dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/filter/AppOverrideFilter.java

```diff
@@ -145,7 +145,11 @@ private void serveInstalledAppResource(
         response.setContentType(mimeType);
       }
 
-      response.setContentLength((int) resource.contentLength());
+      // we need to handle scenarios when the Resource is a File (knowing the content length)
+      // or when it's URL (not knowing the content length and having to make a call, e.g. remote web
+      // link in AWS S3/MinIO) - otherwise content length can be set to 0 which causes issues at
+      // the front-end, returning an empty body.
+      response.setContentLength(appManager.getUriContentLength(resource));
       response.setHeader("ETag", etag);
```

Comment:
- Correct me if I'm wrong but if this `AppOverrideFilter` is triggered for every requests to the server as defined below, then the `getUricontentLength` would be called for every single request.  Do we have a cache mechanism for this value ? Maybe `UrlConnection.useCaches` ? 
```
context
        .addFilter("AppOverrideFilter", new DelegatingFilterProxy("appOverrideFilter"))
        .addMappingForUrlPatterns(null, true, "/*");
```

Reply:
Nice callout. The URL connection uses caching by default 👍
I can update the flow so that we only make the connection if it's not a File though, which might be better (and what the current behaviour is)